src/execution_plan/execution_plan_debug.c

```diff
@@ -31,6 +32,8 @@ void _ExecutionPlan_Print(const OpBase *op, RedisModuleCtx *ctx, sds *buffer,
 void ExecutionPlan_Print(const ExecutionPlan *plan, RedisModuleCtx *ctx) {
 	ASSERT(plan && ctx);
 
+	if(ErrorCtx_EncounteredError()) return;
```

Comment:
Please remove.

In `cmd_explain.c`

```c
ExecutionPlan_PreparePlan(plan);
ExecutionPlan_Init(plan);       // Initialize the plan's ops.
if(ErrorCtx_EncounteredError()) {
    ErrorCtx_EmitException();
    goto cleanup;
}

ExecutionPlan_Print(plan, ctx); // Print the execution plan.
cleanup:
```

Reply:
why?
every time calling to ExecutionPlan_Print it is needed

Reply:
this is mixing boundaries, it is preferred that the execution-plan won't be aware of ErrorCtx,
also it seems weird that a call made to `ExecutionPlan_PreparePlan` won't print anything...